src/main/java/org/jenkinsci/plugins/matrixauth/AuthorizationMatrixNodeProperty.java

```diff
@@ -0,0 +1,176 @@
+    }
+
+    /**
+     * Adds to {@link #grantedPermissions}. Use of this method should be limited
+     * during construction, as this object itself is considered immutable once
+     * populated.
+     */
+    public void add(Permission p, String sid) {
+        Set<String> set = grantedPermissions.get(p);
+        if (set == null)
+            grantedPermissions.put(p, set = new HashSet<String>());
+        set.add(sid);
+        sids.add(sid);
+    }
+
+    @Override
+    public boolean isBlocksInheritance() {
+        return blocksInheritance;
+    }
+
+    @Override
+    public void setBlocksInheritance(boolean blocksInheritance) {
+        this.blocksInheritance = blocksInheritance;
+    }
+
+    private final class AclImpl extends SidACL {
+        @CheckForNull
+        @SuppressFBWarnings(value = "NP_BOOLEAN_RETURN_NULL",
+                justification = "As designed, implements a third state for the ternary logic")
+        protected Boolean hasPermission(Sid sid, Permission p) {
+            if (AuthorizationMatrixNodeProperty.this.hasPermission(toString(sid),p,sid instanceof PrincipalSid)) {
+                return true;
+            }
+            return null;
+        }
+    }
+
+    public SidACL getACL() {
+        return acl;
+    }
+
+    /**
+     * Persist {@link ProjectMatrixAuthorizationStrategy} as a list of IDs that
+     * represent {@link ProjectMatrixAuthorizationStrategy#grantedPermissions}.
+     */
+    public static final class ConverterImpl extends AbstractMatrixPropertyConverter {
+        public boolean canConvert(Class type) {
+            return type == AuthorizationMatrixNodeProperty.class;
+        }
+
+        public AuthorizationProperty createSubject() {
+            return new AuthorizationMatrixNodeProperty();
+        }
+    }
+
+    @Extension
+    public static class DescriptorImpl extends NodePropertyDescriptor implements AuthorizationMatrixPropertyDescriptor<AuthorizationMatrixNodeProperty> {
+
+        @Override
+        public AuthorizationMatrixNodeProperty createProperty() {
+            return new AuthorizationMatrixNodeProperty();
+        }
+
+        @Override
+        public PermissionScope getPermissionScope() {
+            return PermissionScope.COMPUTER;
+        }
+
+        @Override
+        public AuthorizationMatrixNodeProperty newInstance(StaplerRequest req, JSONObject formData) throws FormException {
+            return createNewInstance(req, formData, false);
+        }
+
+        @Override
+        public boolean isApplicable(Class<? extends Node> folder) {
+            return isApplicable();
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Authorization Matrix";
+        }
+
+        public FormValidation doCheckName(@AncestorInPath Computer computer, @QueryParameter String value) throws IOException, ServletException {
+            // TODO Computer needs to become a DescriptorByNameOwner for the AncestorInPath to work
```

Comment:
File a core PR for it?

Reply:
https://github.com/jenkinsci/jenkins/pull/3009